govcd/nsxt_distributed_firewall_test.go

```diff
@@ -0,0 +1,272 @@
+		if vcd.client.Client.APIVCDMaxVersionIs(">= 35.2") {
+			check.Assert(fwUpdated.DistributedFirewallRuleContainer.Values[index].ActionValue, Equals, randomizedFwRuleDefs[index].ActionValue)
+		} else {
+			check.Assert(fwUpdated.DistributedFirewallRuleContainer.Values[index].Action, Equals, randomizedFwRuleDefs[index].Action)
+		}
+
+		for fwGroupIndex := range fwUpdated.DistributedFirewallRuleContainer.Values[index].SourceFirewallGroups {
+			check.Assert(fwUpdated.DistributedFirewallRuleContainer.Values[index].SourceFirewallGroups[fwGroupIndex].ID, Equals, randomizedFwRuleDefs[index].SourceFirewallGroups[fwGroupIndex].ID)
+		}
+
+		for fwGroupIndex := range fwUpdated.DistributedFirewallRuleContainer.Values[index].DestinationFirewallGroups {
+			check.Assert(fwUpdated.DistributedFirewallRuleContainer.Values[index].DestinationFirewallGroups[fwGroupIndex].ID, Equals, randomizedFwRuleDefs[index].DestinationFirewallGroups[fwGroupIndex].ID)
+		}
+
+		// Ensure the same amount of Application Port Profiles are assigned and created
+		check.Assert(len(fwUpdated.DistributedFirewallRuleContainer.Values), Equals, len(randomizedFwRuleDefs))
+		definedAppPortProfileIds := extractIdsFromOpenApiReferences(randomizedFwRuleDefs[index].ApplicationPortProfiles)
+		for _, appPortProfile := range fwUpdated.DistributedFirewallRuleContainer.Values[index].ApplicationPortProfiles {
+			check.Assert(contains(appPortProfile.ID, definedAppPortProfileIds), Equals, true)
+		}
+
+		// Ensure the same amount of Network Context Profiles are assigned and created
+		definedNetContextProfileIds := extractIdsFromOpenApiReferences(randomizedFwRuleDefs[index].NetworkContextProfiles)
+		for _, networkContextProfile := range fwUpdated.DistributedFirewallRuleContainer.Values[index].NetworkContextProfiles {
+			check.Assert(contains(networkContextProfile.ID, definedNetContextProfileIds), Equals, true)
+		}
+	}
+
+	// Cleanup
+	err = fwRules.DeleteAllRules()
+	check.Assert(err, IsNil)
+	// Check that rules were removed
+	newRules, err := vdcGroup.GetDistributedFirewall()
+	check.Assert(err, IsNil)
+	check.Assert(len(newRules.DistributedFirewallRuleContainer.Values) == 0, Equals, true)
+
+	// Cleanup remaining setup
+	_, err = vdcGroup.DisableDefaultPolicy()
+	check.Assert(err, IsNil)
+	_, err = vdcGroup.DeactivateDfw()
+	check.Assert(err, IsNil)
+	err = ipSet.Delete()
+	check.Assert(err, IsNil)
+	err = secGroup.Delete()
+	check.Assert(err, IsNil)
+	err = vdcGroup.Delete()
+	check.Assert(err, IsNil)
+	err = vdc.DeleteWait(true, true)
+	check.Assert(err, IsNil)
+}
+
+// createDistributedFirewallDefinitions creates some randomized firewall rule configurations to match possible configurations
+func createDistributedFirewallDefinitions(check *C, vcd *TestVCD, vdcGroupId string) ([]*types.DistributedFirewallRule, *NsxtFirewallGroup, *NsxtFirewallGroup) {
+	// This number does not impact performance because all rules are created at once in the API
+	numberOfRules := 40
```

Comment:
Just curious, if `40` was chosen for any reason or simply a random number?

Reply:
Totally random. Big enough to generate enough different setups